### PR TITLE
Removed inheritence from SignatureGenerationRule.

### PR DESCRIPTION
**Pull request by adngdb**

### CODE REVIEW

File: socorro/processor/signature_utilities.py [hunk 1]
```diff
@@ -687,41 +687,32 @@ def _get_crashing_thread(self, processed_crash):
...
-        result = super(SignatureShutdownTimeout, self)._action(
-            raw_crash,
-            raw_dumps,
-            processed_crash,
-            processor_meta
-        )
         parts = ['AsyncShutdownTimeout']
         try:
-            shutdown_data = ujson.loads(raw_crash['AsyncShutdownTimeout'])
+            shutdown_data = json.loads(raw_crash['AsyncShutdownTimeout'])
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: Why this change?

File: socorro/processor/signature_utilities.py [hunk 2]
```diff
@@ -732,8 +723,16 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
...
-        processed_crash['signature'] = ' | '.join(parts)
-        return result
+        new_sig = ' | '.join(parts)
+        processor_meta['processor_notes'].append(
+            'Signature replaced with a Shutdown Timeout error, '
+            'was: "{}"'.format(processed_crash['signature'])
+        )
+        processed_crash['signature'] = new_sig
+
+        print 3
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: debugging

File: socorro/processor/signature_utilities.py [hunk 3]
```diff
@@ -687,41 +687,32 @@ def _get_crashing_thread(self, processed_crash):
...
             "shutdownhang | %s" % processed_crash['signature']
         )
-        return result
+        return True
 
 
 #==============================================================================
-class SignatureShutdownTimeout(SignatureGenerationRule):
+class SignatureShutdownTimeout(Rule):
     """Customize the signature for shutdown timeouts"""
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: We should expand this a bit. 
In particular, this doesn't "customize" the signature. It **overwrites the signature**. ...but only if the raw crash contains a truthy value for the key `AsyncShutdownTimeout`.

File: socorro/processor/signature_utilities.py (hunk 1 above)
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: It's plausible that this is more memory efficient than having to load in the binary `ujson`.

File: socorro/processor/signature_utilities.py [hunk 4]
```diff
@@ -732,8 +723,16 @@ def _action(self, raw_crash, raw_dumps, processed_crash, processor_meta):
                 'Error parsing AsyncShutdownTimeout: {}'.format(e)
             )
 
-        processed_crash['signature'] = ' | '.join(parts)
-        return result
+        new_sig = ' | '.join(parts)
+        processor_meta['processor_notes'].append(
+            'Signature replaced with a Shutdown Timeout error, '
```
Comment (peterbe) [inline socorro/processor/signature_utilities.py]: "error"? I struggle to understand this sentence. Should it be "Signature replaced by a Shutdown Timeout signature"?

File: socorro/processor/signature_utilities.py (hunk 1 above)
Comment (adngdb) [reply socorro/processor/signature_utilities.py]: I think we decided at some point to use `json` instead of `ujson`, but I don't remember the reasoning. But looking through the `processor` module, it seems we mostly use ujson so I'll revert that.

Comment (peterbe) [reply socorro/processor/signature_utilities.py]: No, ujson is still the preferred tool. 
There was a time when ujson failed horribly with segfaults. But it was lucky because it was horribly broken raw crash JSON that weighed many gigabytes! Had ujson NOT failed those documents would have been saved (the processed crash would still be small) and cause excess disk usage and would cause problems inside caching etc. 

We used that moment to instead direct our attention to the crashes from breakpad and I think Ted was able to change something so it doesn't happen again. We also put in some code that prevents json documents from being too large in case this ever happens again. 

Lastly, at the time we discovered we ran an ancient version of ujson. We later upgraded. After upgrading I tested putting those Gb raw crashes through ujson and then it did work. 

I do suspect that there are a great deal of cases where the difference between ujson and regular json is 0.0001 seconds. :)
